src/oca_pre_commit_hooks/check_unused_python_file.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+import ast
+import functools
+import os
+from typing import List, Set, Union
+
+
+@functools.lru_cache()
+def get_imported_files(dirname: str) -> Set[str]:
+    imported_files = set()
+    init_file = os.path.join(dirname, "__init__.py")
+    if os.path.exists(init_file):
+        with open(init_file, encoding="utf-8") as init_fd:
+            init_ast = ast.parse(init_fd.read())
+
+        if isinstance(init_ast.body, list):
+            for module in filter(lambda elem: isinstance(elem, ast.ImportFrom), init_ast.body):
+                for name in module.names:
+                    imported_files.add(name.name)
+
+    return imported_files
+
+
+def check_unused_python_file(filenames: List[str]) -> int:
+    status = 0
+    for filename in filenames:
+        if os.path.basename(filename)[:-3] not in get_imported_files(os.path.dirname(filename)):
```

Comment:
In order to avoid duplicating filenames related to the same file but using differente way
e.g. `./module1/fname1.py module1/fname1.py /home/user/module1/fname1.py` and so on

`set(map(utils.full_norm_path, filenames))`

(code no tested)

Related to:
 - https://github.com/OCA/odoo-pre-commit-hooks/blob/6ce447288698e1ade79b773d94e6b8d36f4a63f0/src/oca_pre_commit_hooks/utils.py#L136

Reply:
Interesting, I thought pre-commit dealt with that and returned files in the same way. Do we have a way to reproduce this issue?